include/osmium/io/metadata_options.hpp

```diff
@@ -92,27 +92,7 @@ namespace osmium {
                 m_options = static_cast<options>(opts);
             }
 
-            metadata_options(const osmium::OSMObject& object) {
-                int opts = 0;
-                if (object.version() > 0) {
-                    opts |= options::md_version;
-                }
-                if (object.timestamp().valid()) {
-                    opts |= options::md_timestamp;
-                }
-                if (object.changeset() > 0) {
-                    opts |= options::md_changeset;
-                }
-                // Objects by anonymous users don't have these attributes set. There is no way
-                // to distinguish them from objects with a reduced number of metadata fields.
-                if (object.uid() > 0) {
-                    opts |= options::md_uid;
-                }
-                if (object.user() && strlen(object.user()) > 0) {
-                    opts |= options::md_user;
-                }
-                m_options = static_cast<options>(opts);
-            }
+            friend metadata_options metadata_options_from_object(const osmium::OSMObject& object);
```

Comment:
No friends please. The point of making this an extra function is to decuple internals from functions that don't need it. You'll need to add setter functions to metadata_options for this.